### PR TITLE
BaSP-Feature/MR-135: Forms Activities Classes Subscriptions Trainers

### PR DESCRIPTION
**Pull request by jramire5**
Change shared input. Now it receives 2 new props: register (for handling data in the inputs) and error

Implement Absolute routes for the imports in forms.

Implement react hook forms and joi for the validations in the following entitys: Activities Classes Subscriptions Trainers

Implement react-select library for selects. 

Page should look like last week but now it should display messages below every input if a validation fails.

Add a new reset button that should turn every input blank again in case we are adding a new one and should reset the original data from the entity the user want to edit. (Classes and Trainers)

### CODE REVIEW

File: src/Validations/trainer.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    .min(10)
+    .max(12)
+    .regex(/^[0-9]+$/)
+    .messages({
+      'string.pattern.base': 'Phone must be only numbers',
+      'string.min': 'Phone should have a minimum length of 10',
+      'string.empty': 'Phone is required'
+    })
+    .required(),
+  email: Joi.string().required(),
```
Comment (FontanR) [inline src/Validations/trainer.js]: make a stronger validation for email

File: src/Components/Subscriptions/index.jsx [hunk 2]
```diff
@@ -39,13 +38,14 @@ const Subscriptions = () => {
...
-        <Input
+        <input
+          className={styles.inputSearch}
           name="Search Subscription"
-          labelText="Search"
           placeholder="Search Subscription"
           value={searchTerm}
-          change={(event) => setSearchTerm(event.target.value)}
+          onChange={(event) => setSearchTerm(event.target.value)}
         />
```
Comment (FontanR) [inline src/Components/Subscriptions/index.jsx]: if you use a placeholder the label is not needed

**Review by FontanR — COMMENTED** [review]
Try to be more consistent with the messages from the validations
Besides that, everything works great
Excellent job!

**Review by FontanR — APPROVED** [review]

File: src/Validations/trainer.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      'string.max': 'Last Name should have a maximum length of 10',
+      'string.empty': 'Last Name is required'
+    }),
+  dni: Joi.string()
+    .min(7)
+    .max(10)
+    .regex(/^[0-9]+$/)
+    .message({
+      'string.pattern.base': 'DNI should be only numbers',
+      'string.empty': 'Dni is required'
```
Comment (josedaniele) [inline src/Validations/trainer.js]: Write DNI to maintain consistency with the message above

File: src/Components/Classes/List/ClassList.jsx [hunk 4]
```diff
@@ -1,12 +1,12 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom/cjs/react-router-dom.min';
 import styles from './list.module.css';
 import Button from '../../Shared/Button';
 import ConfirmModal from '../../Shared/ConfirmModal';
-import { Input } from '../../Shared/Inputs';
 import { useDispatch } from 'react-redux';
 import { deleteClass } from '../../../Redux/Classes/thunks';
 import { refreshData } from '../../../Redux/Classes/actions';
```
Comment (josedaniele) [inline src/Components/Classes/List/ClassList.jsx]: you could use absolute paths

**Review by josedaniele — CHANGES_REQUESTED** [review]

**Review by josedaniele — APPROVED** [review]

**Pull request merged**